### PR TITLE
Like button click: backend and UI changes

### PR DESCRIPTION
**Pull request by pravagar**
- Reused metric controller PUT api which is used to update download count for Like/dislike user action.
- Existing UpdateCounts function app updated to take care of Like count update.
- Added a new field named Likes in Library controller.

### CODE REVIEW

File: ServerlessLibraryAPI/Controllers/MetricsController.cs [hunk 1]
```diff
@@ -9,13 +9,13 @@ namespace ServerlessLibrary.Controllers
     [ApiController]
     public class MetricsController : ControllerBase
     {
-        // PUT api/<controller>/
+        // PUT api/<controller>/{userAction}
         [ProducesResponseType(typeof(bool), 200)]
-        [HttpPut()]
-        public JsonResult Put([FromBody]string template)
+        [HttpPut]
+        [Route("{userAction}")]
```
Comment (msnehagup) [inline ServerlessLibraryAPI/Controllers/MetricsController.cs]: why do you need to add this additional route attrib?
PUT on /api/metrics should work without this additional route attrib

File: ServerlessLibraryAPI/StorageHelper.cs [hunk 2]
```diff
@@ -71,25 +71,26 @@ public static async Task<IEnumerable<SLItemStats>> getSLItemRecordsAsync()
...
 
- }
+    }
 
     public class SLItemStats : TableEntity
     {
-        public string template { get; set; }
+        public string id { get; set; }
         public int totalDownloads { get; set; }
         public int downloadsToday { get; set; }
```
Comment (msnehagup) [inline ServerlessLibraryAPI/StorageHelper.cs]: are we still going to keep downloads today/this week/this month ? do we show them anywhere?

Comment (pravagar) [reply ServerlessLibraryAPI/StorageHelper.cs]: In new design, not used. I am open to remove these.

File: ServerlessLibraryAPI/Controllers/MetricsController.cs (hunk 1 above)
Comment (pravagar) [reply ServerlessLibraryAPI/Controllers/MetricsController.cs]: yes. In last iteration, this was used. Now no longer required. will remove

File: ServerlessLibraryAPI/StorageHelper.cs (hunk 2 above)
Comment (pravagar) [reply ServerlessLibraryAPI/StorageHelper.cs]: removed

File: ServerlessLibraryAPI/Controllers/MetricsController.cs (hunk 1 above)
Comment (pravagar) [reply ServerlessLibraryAPI/Controllers/MetricsController.cs]: removed

File: ServerlessLibraryAPI/ClientApp/src/services/library.service.js [hunk 3]
```diff
@@ -37,3 +35,17 @@ function submitNewSample(item) {
...
+function updateUserSentimentStats(id, likeChanges, dislikeChanges) {
+  const requestOptions = {
+    method: "PUT",
+    body: '"' + id + '"',
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+  return fetch(
+    "/api/metrics/sentiment/" + likeChanges + "/" + dislikeChanges,
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/services/library.service.js]: IMHO the likes and dislikes should go into the body

File: ServerlessLibraryFunctionApp/UpdateCounts.cs [hunk 4]
```diff
@@ -17,56 +17,106 @@ public static class UpdateCounts
...
                 var item = entities[0];
-                item.downloadsThisMonth += 1;
-                item.downloadsThisWeek += 1;
-                item.downloadsToday += 1;
-                item.totalDownloads += 1;
-                TableOperation operation = TableOperation.InsertOrMerge(item) ;
+                switch (userAction)
+                {
+                    case UserAction.Download:
+                        item.downloadsThisMonth += 1;
```
Comment (msnehagup) [inline ServerlessLibraryFunctionApp/UpdateCounts.cs]: totalDownloads

Comment (pravagar) [reply ServerlessLibraryFunctionApp/UpdateCounts.cs]: done

File: ServerlessLibraryAPI/ClientApp/src/services/library.service.js (hunk 3 above)
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/services/library.service.js]: yup

File: ServerlessLibraryAPI/ClientApp/src/components/Main/Main.js [hunk 5]
```diff
@@ -23,7 +23,6 @@ class Main extends Component {
       el =>
         el.title.match(filter) ||
         el.description.match(filter) ||
-        el.authortype.match(filter) ||
         el.language.match(filter) ||
         el.type.match(filter) ||
         el.repository.replace("https://github.com/", "").match(filter) ||
```
[the remark this answers is not in the document]
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/Main/Main.js]: yes, in my opinion :)

File: ServerlessLibraryAPI/ClientApp/src/components/MetricBar/MetricBar.js [hunk 6]
```diff
@@ -42,21 +42,29 @@ class MetricBar extends Component {
...
     libraryService
-      .updateUserSentimentStats(this.props.id, likeChanges, dislikeChanges)
+      .updateUserSentimentStats(sentimentPayload)
       .then(response => response.body)
       .catch(error => console.log(error));
   }
 
   render() {
     let {author, downloads, createddate, likes, dislikes} = this.props;
+    console.log(createddate);
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/MetricBar/MetricBar.js]: remove :)

**Pull request merged**